File: app/src/main/java/org/dukecon/android/features/main/di/MainComponent.java
package org.dukecon.android.features.main.di;

import org.dukecon.android.app.di.ActivityScope;
import org.dukecon.android.features.sessions.di.SessionDatesComponent;
import org.dukecon.android.features.sessions.di.SessionDatesModule;
import org.dukecon.android.features.sessions.di.SessionListComponent;
import org.dukecon.android.features.sessions.di.SessionsListModule;

import dagger.Subcomponent;

@ActivityScope
@Subcomponent(
        modules = {MainModule.class}
)
public interface MainComponent {

    SessionDatesComponent plus(SessionDatesModule loginModule);

    SessionListComponent plus(SessionsListModule sessionsListModule);
}

File: mobile-ui/src/main/java/org/dukecon/android/ui/features/main/di/MainComponent.kt
package org.dukecon.android.ui.features.main.di

import org.dukecon.android.ui.features.event.di.EventsListComponent
import dagger.Subcomponent

@Subcomponent(modules = arrayOf(MainModule::class))
interface MainComponent {
    fun sessionListComponent(): EventsListComponent
//    fun speakerListComponent(): SpeakerListComponent
//    fun locationComponent(): LocationComponent
}